sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -583,7 +583,9 @@ object FunctionRegistry {
           val validParametersCount = constructors
             .filter(_.getParameterTypes.forall(_ == classOf[Expression]))
             .map(_.getParameterCount).distinct.sorted
-          val expectedNumberOfParameters = if (validParametersCount.length == 1) {
+          val expectedNumberOfParameters = if (validParametersCount.length == 0) {
+            throw new AnalysisException(s"$name is an invalid function.")
```

Comment:
Yea, I agree with https://github.com/apache/spark/pull/24947#issuecomment-505260696. Let's just raise an exception like this. It's an invalid function if there's no constructor to call anyway.